### ISSUE TITLE
Prevent chiventure from starting in terminals smaller than 80 columns x 24 rows

### ISSUE THREAD

**Issue by borjasotomayor** [uchicago-cs/chiventure#415]
While games can technically be run in terminals with less than 80 columns and less than 24 rows, it will result in excessively wrapped text, and will also complicate displaying the map (as well as complicating future features that assume a screen of some minimum size). chiventure should refuse to run if the terminal doesn't have the required size. To futureproof this decision, it should be easy to later on change this minimum size, if needed.

**Comment by jackandthebean** [comment]
In the `ui/terminal-dimensions` branch, chiventure now refuses to run if the terminal doesn't have the required size. The minimum dimensions are easily reconfigured, as macros `MIN_COLS` and `MIN_ROWS` are now defined in `ui.h`.

5bd7e3ca9cecb63673ad63bb51b26e78ec103ed8

I consulted [this](https://psychocod3r.wordpress.com/2019/02/25/how-to-get-the-dimensions-of-a-linux-terminal-window-in-c/) tutorial to obtain the dimensions of the terminal.

I will now move on to handling the case where a user resizes a previously valid terminal to one of invalid size, after a chiventure game is begun.

**Comment by jackandthebean** [comment]
The branch I created to make chiventure refuse to run if the terminal doesn't have the required dimensions has now been merged into `dev`, so I am closing this issue! However, it seems like a good idea to also implement a way to put chiventure on hold if after starting successfully in a terminal of the correct dimensions, a user resizes their terminal to unsupported dimensions. I added an issue describing this enhancement to the backlog – #650. If I have time I may also take on this issue in Sprint 4; otherwise, anyone can claim this enhancement!
